controller/src/server/functions.ts

```diff
@@ -268,10 +285,33 @@ export async function queueRun(run_id:string):Promise<string> {
   return 'ok';
 }
 
+function connectHasuraEndpoint():void {
+  const hasura = process.env.HASURA ?? 'http://127.0.0.1:8080/v1/graphql';
+  client = new GraphQLClient(hasura, {
+    headers: {
+      'x-hasura-admin-secret': 'hasuraadminsecret',
+    },
+  });
+  sdk = getSdk(client);
+}
+
 export async function createSampleSimulation():Promise<string> {
-  // check if there are simulations in the database
-  const result = await allSimulations();
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  await setTimeout(7000);
```

Comment:
Why do we wait 7 seconds ?

Reply:
There are 2 main steps when the  controller starts up:
1. starts the graphql API
2. run createSampleSImulation function which creates a sample simulation in case there are no existing simulations (mainly for the user to play with)

Hasura is dependent on graphql API because it is the remote schema hasura connects to.
So hasura can start up only after controller starts up. This order is already specified in docker-compose. 

But step 2 in controller start up depends on hasura to access the database. Hence there is a delay in between steps 1 and 2 to wait for the hasura to start up.